inc/sitemaps/class-sitemaps-cache.php

```diff
@@ -41,6 +47,190 @@ public function is_enabled() {
+
+			$type = $first_part . '..' . $last_part;
+		}
+
+		return $type;
+	}
+
+	/**
+	 * Get the cache validator for the specified type
+	 *
+	 * @param string|null $type Provide a type for a specific type validator, null for global validator.
+	 *
+	 * @return string Validator to be used to generate the cache key.
+	 */
+	public static function get_validator_key( $type = null ) {
+		if ( is_null( $type ) ) {
+			return 'wpseo_sitemap_cache_validator_global';
+		}
+
+		return sprintf( 'wpseo_sitemap_%s_cache_validator', $type );
+	}
+
+	/**
+	 * Get the current cache validator
+	 *
+	 * Without the type the global validator is returned.
+	 *  This can invalidate -all- keys in cache at once
+	 *
+	 * With the type parameter the validator for that specific
+	 *  type can be invalidated
+	 *
+	 * @param string|null $type Provide a type for a specific type validator, null for global validator.
+	 *
+	 * @return null|string The validator for the supplied type.
+	 */
+	private static function get_validator( $type = null ) {
+		$key = self::get_validator_key( $type );
+
+		$current = get_option( $key, null );
+		if ( ! is_null( $current ) ) {
+			return $current;
+		}
+
+		if ( self::new_validator( $type ) ) {
+			return self::get_validator( $type );
+		}
+
+		return null;
+	}
+
+	/**
+	 * Refresh the cache validator value
+	 *
+	 * @param string|null $type Provide a type for a specific type validator, null for global validator.
+	 *
+	 * @return bool True if validator key has been saved as option.
+	 */
+	public static function new_validator( $type = null ) {
+		$key = self::get_validator_key( $type );
+
+		// Generate new validator.
+		$microtime = microtime();
+
+		// Remove space.
+		list( $milliseconds, $seconds ) = explode( ' ', $microtime );
+
+		// Transients are purged every 24h.
+		$seconds      = ( $seconds % DAY_IN_SECONDS );
+		$milliseconds = substr( $milliseconds, 2, 5 );
+
+		// Combine seconds and milliseconds and convert to integer.
+		$validator = intval( $seconds . '' . $milliseconds, 10 );
+
+		// Apply base 61 encoding.
+		$compressed = self::convert_base10_to_base61( $validator );
+
+		return update_option( $key, $compressed );
+	}
+
+	/**
+	 * Encode to base61 format.
+	 *
+	 * This is base64 (numeric + alpha + alpha upper case) without the 0.
+	 *
+	 * @param int $input The number that has to be converted to base 61.
+	 *
+	 * @return string Base 61 converted string.
+	 *
+	 * @throws InvalidArgumentException When the input is not an integer.
+	 */
+	public static function convert_base10_to_base61( $input ) {
+		if ( ! is_int( $input ) ) {
+			throw new InvalidArgumentException( __( 'Expected an integer as input.', 'wordpress-seo' ) );
+		}
+
+		$characters = '123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+		$length     = strlen( $characters );
+
+		$index  = ( $input % $length );
+		$output = $characters[ $index ];
+
+		$position = floor( $input / $length );
```

Comment:
This is not a function you can create without inline comments.

For example you could describe working from 'most significant' to 'least significant' or vice versa.